src/flick/flick_auth/serializers.py

```diff
@@ -38,54 +20,9 @@ class Meta:
         )
         write_only_fields = fields
 
-    def save(self, **kwargs):
-        data = self.init_data if hasattr(self, "init_data") else self.initial_data
-
-        items = dict(data.items())
-        user_data = {
-            "username": items["username"],
-            "first_name": items["first_name"],
-            "last_name": items["last_name"],
-            "password": items["social_id_token"],
-        }
-
-        profile_data = {
-            "profile_pic": items["profile_pic"],
-            "social_id_token": items["social_id_token"],
-            "social_id_token_type": items["social_id_token_type"],
-            "role": "consumer",
-        }
-
-        group = profile_data["role"] + "_basic"
-        user = AuthTools.register(user_data, profile_data, group)
-
-        if user is not None:
-            self.object = user
-            return self.object
-
-        raise serializers.ValidationError("Unable to register with the credentials provided.")
-
 
 class LoginSerializer(serializers.Serializer):
```

Comment:
kinda confused by the implementation, does it need to have a source? and seems like auth token is passed into the serializer. Not sure how that is used?

Reply:
this is the old code, so not exactly sure what you mean by it needing a source -- and yes the auth token is displayed through the serializer :)